### PR TITLE
Issue 2 user can view starting11 - Aaditya Desai

### PR DESCRIPTION
**Pull request by adesai47**
Recording 2024-06-27 at 09 26 05

### CODE REVIEW

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Looks pretty good but I the player boxes should be a consistent size and spell out their positions (e.g. D -> Defense, GK -> Goalkeeper. . .)

**Comment by adesai47** [comment]
Screenshot 2024-06-27 at 10 23 34 AM

File: ts/main.ts [hunk 1]
```diff
@@ -1,65 +1,171 @@
...
+    });
+  }
 
-fetchTeamLogos(frlApiUrl, ['Paris Saint-Germain FC']);
+  loadTeams();
+});
+
+function showPage(pageId: string): void {
+  console.log(`Showing page: ${pageId}`);
+  const pages = document.querySelectorAll('main section');
```
Comment (brandon-moy) [inline ts/main.ts]: Remove the console.log here

File: ts/main.ts [hunk 2]
```diff
@@ -1,65 +1,171 @@
...
+    console.log(`Fetched players for ${teamName}:`, players);
+    starting11Container.innerHTML = '';
 
-        teamElement.appendChild(teamLogoElement);
-        teamElement.appendChild(teamNameElement);
+    players.forEach((player: Player) => {
+      const playerElement = document.createElement('div');
+      playerElement.classList.add('player');
+      playerElement.innerHTML = `
+        <div class="player-jersey">#${player.Jersey}</div>
```
Comment (brandon-moy) [inline ts/main.ts]: Do not use `innerHTML`, you should be creating the elements and appending them.

File: ts/main.ts [hunk 3]
```diff
@@ -1,65 +1,171 @@
...
+  if (!apiUrl || !starting11Container) return;
 
-        const teamNameElement: HTMLElement = document.createElement('div');
-        teamNameElement.classList.add('team-name');
-        teamNameElement.innerHTML = `<p>${team.Name}</p>`;
+  try {
+    const response: Response = await fetch(apiUrl);
+    const players: Player[] = await response.json();
+    console.log(`Fetched players for ${teamName}:`, players);
+    starting11Container.innerHTML = '';
```
Comment (brandon-moy) [inline ts/main.ts]: Remove the console.log and you should remove a different way if needed without using `innerHTML`

File: ts/main.ts [hunk 4]
```diff
@@ -1,65 +1,171 @@
...
+          <div class="team-logo">
+            <img src="${team.WikipediaLogoUrl}" alt="${team.Name}">
+          </div>
+          <div class="team-name"><p>${team.Name}</p></div>
+        `;
+        teamElement
+          .querySelector('.team-logo')
+          ?.addEventListener('click', () => {
+            console.log(`Team clicked: ${team.Name}`);
+            if (teamLogoElement) {
```
Comment (brandon-moy) [inline ts/main.ts]: Remove console.log

File: ts/main.ts [hunk 5]
```diff
@@ -1,65 +1,171 @@
...
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/EPL/517?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Liverpool FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/EPL/515?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'FC Barcelona':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/ESP/559?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Real Madrid CF':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/ESP/605?key=d96ee95088b9438aa8c07a785c81b6e9',
+  'Paris Saint-Germain FC':
+    'https://api.sportsdata.io/v4/soccer/scores/json/PlayersByTeamBasic/FRL1/600?key=d96ee95088b9438aa8c07a785c81b6e9',
+};
```
Comment (brandon-moy) [inline ts/main.ts]: Is there any way you can do this without having all of your endpoints hard coded? This makes your project very closed off in only being what you decide which may not be what a user wants.

I see that it does change the teams or leagues with the EPL, ESP, or FRL!, but can you get the number after some other way? Does it come from your first API call and you can attach it to an element and access it that way?

**Review by brandon-moy — CHANGES_REQUESTED** [review]
More changes requested from your code and not just the images you provided.

**Review by brandon-moy — APPROVED** [review]
Approved 👍

**Pull request merged**